client/institution.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/mxenabled/atrium-go/models"
+	"net/http"
+)
+
+
+func parseInstitutionResponse(response *http.Response) (*models.Institution, error) {
+	if err := parseResponseErrors(response.StatusCode); err != nil {
+		return nil, err
+	}
+
+	buffer := new(bytes.Buffer)
+	buffer.ReadFrom(response.Body)
+	bufferStr := buffer.String()
+	response.Body.Close()
+
+	if response.StatusCode == 200 {
+		institutionResponse := &models.InstitutionResponse{}
+		json.Unmarshal([]byte(bufferStr), institutionResponse)
+		return institutionResponse.Institution, nil
+	}
+
+	return nil, makeGenericError(response.StatusCode, bufferStr)
+}
+
+func (c *Client) ListInstitutions(Name string) ([]*models.Institution, error) {
+	var param = "?"
+	if Name != "" {
+		param += "name=" + Name + "&"
+	}
+	param = param[:len(param)-1]
```

Comment:
I think you can use `"net/url"` package to do:
```go
params := url.Values{}
if len(name) > 0 {
  params.Add("name", name)
}
paramsStr := params.Encode()
```
https://golang.org/pkg/net/url/#Values